paddle/phi/kernels/kps/reduce_sum_kernel.cu

```diff
@@ -29,10 +78,53 @@ void SumRawKernel(const Context& dev_ctx,
   if (out_dtype == DataType::UNDEFINED && out->dtype() != x.dtype()) {
     out_dtype = out->dtype();
   }
-  phi::Reduce<T, kps::AddFunctor, kps::IdentityFunctor>(
-      dev_ctx, x, reduce_all, dims.GetData(), keep_dim, out_dtype, out);
-}
+  if (x.numel() > INT_MAX) {
+#ifndef PADDLE_WITH_XPU_KP
```

Comment:
Seems that `out_dtype` is not considered here. How about raise an exception directly if a valid `out_dtype` is provided?